monai/data/dataset.py

```diff
@@ -74,8 +76,8 @@ def _transform(self, index: int):
     def __getitem__(self, index: int):
```

Comment:
The type of `index` and the return type do need to be adjusted now. The documentation should also state that the return value relies on whether `index` is a single value or a slice.

Reply:
We should also note explicitely that our behaviour is not diverging from Pytorch's `Dataset` class in this specific way, up to now the two were substitutable.

Reply:
Sure, let me update the doc-strings.
Thanks.

Reply:
Hi @ericspod , added doc-strings for this enhancement: https://github.com/Nic-Ma/MONAI/blob/1821-add-slicing-datasets/monai/data/dataset.py#L48

Thanks.